nemo/collections/asr/models/confidence_ensemble.py

```diff
@@ -0,0 +1,203 @@
+
+    def setup_training_data(self, train_data_config: Union[DictConfig, Dict]):
+        """Pass-through to the ensemble models.
+
+        Note that training is not actually supported for this class!
+        """
+        for model_idx in range(self.num_models):
+            getattr(self, f"model{model_idx}").setup_training_data(train_data_config)
+
+    def setup_validation_data(self, val_data_config: Union[DictConfig, Dict]):
+        """Pass-through to the ensemble models."""
+        for model_idx in range(self.num_models):
+            getattr(self, f"model{model_idx}").setup_validation_data(val_data_config)
+
+    def change_attention_model(
+        self, self_attention_model: str = None, att_context_size: List[int] = None, update_config: bool = True
+    ):
+        """Pass-through to the ensemble models."""
+        for model_idx in range(self.num_models):
+            getattr(self, f"model{model_idx}").change_attention_model(
+                self_attention_model, att_context_size, update_config
+            )
+
+    def change_decoding_strategy(self, decoding_cfg: DictConfig = None, decoder_type: str = None):
+        """Pass-through to the ensemble models.
+
+        The only change here is that we always require frame-confidence to
+        be returned.
+        """
+        decoding_cfg.confidence_cfg = self.confidence_cfg
+        for model_idx in range(self.num_models):
+            model = getattr(self, f"model{model_idx}")
+            if isinstance(model, EncDecHybridRNNTCTCModel):
+                model.change_decoding_strategy(decoding_cfg, decoder_type=decoder_type)
+            else:
+                model.change_decoding_strategy(decoding_cfg)
+
+    @torch.no_grad()
+    def transcribe(
+        self,
+        paths2audio_files: List[str],
+        batch_size: int = 4,
+        return_hypotheses: bool = False,
+        num_workers: int = 0,
+        channel_selector: Optional[ChannelSelectorType] = None,
+        augmentor: DictConfig = None,
+        verbose: bool = True,
+        **kwargs,  # any other model specific parameters are passed directly
+    ) -> List[str]:
+        """Confidence-ensemble transcribe method.
+
+        Consists of the following steps:
+
+            1. Run all models (TODO: in parallel)
+            2. Compute confidence for each model
+            3. Use logistic regression to pick the "most confident" model
+            4. Return the output of that model
+        """
+        # TODO: lots of duplicate code with building ensemble script
+        aggr_func = get_confidence_aggregation_bank()[self.confidence_cfg.aggregation]
+        confidences = []
+        all_transcriptions = []
+        # always requiring to return hypothesis
+        # TODO: make sure to return text only if was False originally
+        return_hypotheses = True
+        for model_idx in range(self.num_models):
+            model = getattr(self, f"model{model_idx}")
+            transcriptions = model.transcribe(
```

Comment:
will this support transcribe_partial_audio just like `transcribe_speech.py`?

Reply:
Not at the moment, right now the assumption is that we transcribe the full thing and then get the confidence. But in principal, there is a plan to add support (likely with a subclass) that will transcribe only part of the utterance and then continue with the transcription of only the single most-confident model

Reply:
Gotcha. Yeah theoretically all the ASR model (with extended feature) should support transcribing whole utterance and part of it. Please make sure to add this later. Looking forward to it!